### PR TITLE
Recording player meta description card

### PR DESCRIPTION
**Pull request by alexkim205**
## Changes

Implements recordings person meta data card - https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=5084%3A30911

Screen Shot 2021-11-18 at 4 40 55 PM

Loading State:

https://user-images.githubusercontent.com/13460330/142519307-542e9e20-55bb-4084-9d80-e11573ec4118.mov

I think this is the last of the work that needs to be merged in before we release the new events list. After this PR's merged, we should turn on the `new-sessions-player-events-list` FF.

## How did you test this code?

Added new metaLogic.test.ts

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/metaLogic.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        sessionPerson: [
+            (selectors) => [selectors.sessionPlayerData],
+            (playerData): Partial<PersonType> => {
+                return playerData?.person
+            },
+        ],
+        description: [
+            (selectors) => [selectors.sessionPerson],
+            (person) => {
+                const location = getPersonProperties(person, ['$geoip_city_name', '$geoip_country_code'])
```
Comment (rcmarron) [inline frontend/src/scenes/session-recordings/player/metaLogic.ts:35]: We should probably handle the case where we don't have these properties. (e.g. they don't have the geo plugin running)

File: frontend/src/scenes/session-recordings/player/metaLogic.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                return snapshots[index]
+            },
+        ],
+        resolution: [
+            (selectors) => [selectors.currentSnapshot],
+            (snapshot) => {
+                if (!(snapshot && 'width' in snapshot.data && 'height' in snapshot.data)) {
+                    return null
+                }
+                return {
```
Comment (rcmarron) [inline frontend/src/scenes/session-recordings/player/metaLogic.ts:56]: It's awesome how the resolution updates as the user resizes their window!

File: frontend/src/scenes/session-recordings/player/PlayerMeta.tsx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                    <span className="email">
+                        <PersonHeader person={sessionPerson} withIcon={false} />
+                    </span>
+                </Row>
+                <Col>
+                    {sessionPlayerDataLoading ? (
+                        <Skeleton title={false} active paragraph={{ rows: 1, width: 80 }} />
+                    ) : (
+                        <span className="time text-small">
+                            {meta.startTime ? dayjs(meta.startTime).fromNow() : 'a while ago'}
```
Comment (rcmarron) [inline frontend/src/scenes/session-recordings/player/PlayerMeta.tsx:35]: Let's leave it blank if we don't have a start time

**Review by rcmarron — APPROVED** [review]
Looks great - two small comments

File: frontend/src/scenes/session-recordings/player/metaLogic.ts (hunk 1 above)
Comment (alexkim205) [reply frontend/src/scenes/session-recordings/player/metaLogic.ts:35]: Yup I already handle this by returning `null` in `getPersonProperties` if both properties don't exist on `person.properties` and omitting null values in line 37.

**Review by clarkus — APPROVED** [review]
Love the use of skeleton states.

File: frontend/src/scenes/session-recordings/player/metaLogic.ts (hunk 1 above)
Comment (rcmarron) [reply frontend/src/scenes/session-recordings/player/metaLogic.ts:35]: Ah nice! Missed that!

**Comment by alexkim205** [comment]
Two changes

- [x] Fixed the meta loading state so that we don't have to wait until all snapshots have loaded before showing the meta.
- [x] Add avatar circle skeleton for completeness

**Pull request merged**